docs/testplan/Smartswitch-test-plan.md

```diff
@@ -645,6 +672,672 @@ root@sonic:/home/cisco# show chassis modules status
  *  Verify number of DPUs from inventory file for the testbed and number of DPUs shown in the cli output.
 
 
+### 1.13 Check Memory on DPU
+
+### Steps
+
+ * Infrastructure support will be provided to choose from following clis or in combinations based on the vendor.
+ * Use command `show system-memory` to get memory usage on each of those DPUs
+ * Use `show system-health dpu <DPU_NUM>` to check memory check service status
+ * Use `pdsctl show system --events` to check memory related events triggered.
+   (This is vendor specific event monitoring cli) 
+
+#### Verify in
+ * DPU
+   
+#### Sample Output
+```
+On DPU:
+
+root@sonic:/home/admin# show system-memory
+               total        used        free      shared  buff/cache   available
+Mem:            6266        4198        1509          28         765        2067
+Swap:              0           0           0
+root@sonic:/home/admin# 
+root@sonic:/home/admin# 
+root@MtFuji:/home/cisco# show system-health dpu DPU0
+is_smartswitch returning True
+Name    ID    Oper-Status    State-Detail             State-Value    Time               Reason
+------  ----  -------------  -----------------------  -------------  -----------------  --------------------------------------------------------------------------------------------------------------------------
+DPU0    0     Online         dpu_midplane_link_state  UP             20241003 17:57:22  INTERNAL-MGMT : admin state - UP, oper_state - UP, status - OK, HOST-MGMT : admin state - UP, oper_state - UP, status - OK
+                             dpu_control_plane_state  UP             20241003 17:57:22  All containers are up and running, host-ethlink-status: Uplink1/1 is UP
+                             dpu_data_plane_state     UP             20241001 19:54:30  DPU container named polaris is running, pdsagent running : OK, pciemgrd running : OK
+root@MtFuji:/home/cisco# 
+
+root@sonic:/home/admin# pdsctl show system --events
+----------------------------------------------------------------------------------------------------
+Event                                             Severity  Timestamp                               
+----------------------------------------------------------------------------------------------------
+DSE_SERVICE_STARTED                               DEBUG     2024-09-20 21:48:11.515551 +0000 UTC    
+DSE_SERVICE_STARTED                               DEBUG     2024-09-20 21:48:11.668685 +0000 UTC    
+DSE_SERVICE_STARTED                               DEBUG     2024-09-20 21:48:12.379261 +0000 UTC    
+DSE_SERVICE_STARTED                               DEBUG     2024-09-20 21:48:19.379819 +0000 UTC        
+root@sonic:/home/admin#
+
+```
+
+#### Pass/Fail Criteria
+
+ * Verify that used memory should not cross the specified threshold value (90) of total memory.
+ * Threshold can be set different based on platform.
```

Comment:
Where can this threshold be set? will it be REDIS database or platform.json?

Reply:
It can be in platform.json.

Reply:
Do we have this change or PR to set the threshold?